src/eva/data/ioda_obs_space.py

```diff
@@ -0,0 +1,96 @@
+        datasets = config.get('datasets')
+
+        # Loop over the datasets
+        for dataset in datasets:
+
+            # Get the groups to be read
+            groups = get(dataset, logger, 'groups')
+
+            # Assert the correct use of variables and channels
+            if ('variables' in dataset and 'channels' in dataset):
+                self.logger.abort('Dataset configuraiton for \'' + dataset['name'] + '\'' +
+                                  'contains both channels or variables, should be one or ' +
+                                  'the other. If channels present, variable is ' +
+                                  'brightness_temperature')
+
+            # Get channels or variables
+            if ('variables' in dataset):
+                variables = dataset['variables']
+            elif ('channels' in dataset):
+                channels = dataset['channels']
+                variables = ['brightness_temperature']  # If channels present variable is bt
+            else:
+                self.logger.abort('Dataset configuraiton for \'' + dataset['name'] + '\'' +
+                                  'contains neither channels or variables.')
+
+            # Get metadata
+            metadata_variables = dataset.get('metadata', [])
+
+            # Add metadata group to list
+            if metadata_variables:
+                groups.append('MetaData')
+
+            # Filenames to be read into this collection
+            filenames = get(dataset, logger, 'filenames')
+
+            # Loop over groups
+            for group in groups:
+
+                # Set the variables
+                variables_need = variables
+                if group is 'MetaData':
+                    variables_need = metadata_variables
+
+                # Set the collection name
+                collection_name = dataset['name'] + '::' + group
+
+                # Loop over filenames
+                for filename in filenames:
+
+                    # Open file and add collection
+                    with xr.open_dataset(filename, group=group) as ds:
+
+                        # Drop data variables not in user requested variables
+                        vars_to_remove = list(set(list(ds.keys())) - set(variables_need))
+                        ds = ds.drop_vars(vars_to_remove)
+
+                        # Assert that the collection contains at least one variable. If not it is
+                        # likely some problem has occurred
+                        if not ds.keys():
+                            self.logger.abort('Collection \'' + dataset['name'] + ', group ' +
+                                              group + '\' in file ' + filename +
+                                              ' does not have any variables.')
+
+                        # Add the dataset to the collections
+                        data_collections.create_or_add_to_collection(collection_name, ds, 'nlocs')
+
+            print(data_collections)
```

Comment:
Is this for debugging or intended to always print?

Reply:
I think this would be a useful thing to print because it shows the users exactly what variables and collections are available. This would help with debugging the next step if the incorrect collection name or variable is chosen.